### PR TITLE
Add push upgrade for event offline receipt

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Add push upgrade for event offline receipt

This follows our historical practice of doing push receipts by
- adding the relevant receipt to the array of template in the upgrade notification and
- (optionally) noting deprecated params on CRM_Utils_Token::getDeprecatedTokens

We may do additional messaging in the upgrade script around the text version but it should not replace our 'normal' approach

Before
----------------------------------------
Although the event offline template has been re-written to no longer rely on the form the user has not yet been advised to update any cusomisations & we have not yet removed the form code

After
----------------------------------------
User gets push update notification to update their template

The additions to the deprecated tokens will create some pushiness via the system check- but it's mostly to keep a record

Note however, we will see the text version of the offline template triggering the system checks until we remove the text version

I didn't add lineItem to the deprecated array as it is pretty legit to assign that within a foreach loop.
I removed the assigns for the variables which are no longer relevant - this can be followed up with more tidy up (some of which resolves php8.x issues)

Technical Details
----------------------------------------
Note that the template still uses
$billingName & $address - resolved in
https://github.com/civicrm/civicrm-core/pull/27692 $customGroup - resolved in https://github.com/civicrm/civicrm-core/pull/27596 (actually the variable remains but becomes non-leaky) $credit_card_date,$credit_card_type (I propose moving to a paymet token) $credit_card_expiry (I propose removing)
$event.confirm_text - addressed in https://github.com/civicrm/civicrm-core/pull/27162

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Upgrade/Incremental/MessageTemplates.php [hunk 1]
```diff
@@ -359,6 +359,15 @@ protected function getTemplateUpdates() {
           ['name' => 'petition_sign', 'type' => 'subject'],
         ],
       ],
+      [
+        'version' => '5.67.alpha1',
```
Comment (demeritcowboy) [inline CRM/Upgrade/Incremental/MessageTemplates.php:363]: 5.68.alpha1, 'cuz global inflation.

Comment (eileenmcnaughton) [reply CRM/Upgrade/Incremental/MessageTemplates.php:363]: Lol - we have 12% inflation per year on an ongoing basis!

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
@demeritcowboy @colemanw  this one is passing now - there are a bunch of variables that can be cleaned up once this is merged as follow ups

**Comment by demeritcowboy** [comment]
ok will take a look

**Comment by demeritcowboy** [comment]
When you upgrade even if you haven't customized the template it doesn't blank out the text one. I haven't looked but I'm guessing somewhere in the upgrade code it does an `if (!empty($text_from_the_xml_message_template_file))` before updating.

If I put just a space in the tpl then it does update it - I thought it then wouldn't autopopulate in the actual email and would just output a space, but it does autopopulate. I'm not sure if that should be considered a separate bug or not.

Otherwise looks ok. And I suppose the update bug is really a bug in the [previous PR](https://github.com/civicrm/civicrm-core/commit/45a6a7b5a0cdca6ec7c268ee733073c3fae79c27) that "blonked" it out. So, merging.

**Comment by eileenmcnaughton** [comment]
@demeritcowboy thanks - getting this merged gets rid of all those assigns - so yay

On the upgrade - it is intentional that autopopulate happens if the template is empty-like - but it should do the blanking out for unmodified so I'll look at that

**Pull request merged**